lib/Optimizer/GraphOptimizer.cpp

```diff
@@ -855,6 +855,92 @@ static void optimizeBatchNorm(Function *F) {
   } // For all nodes in the graph.
 }
 
+/// Check if the \p inputs of this method (vector of NodeValues) can be
+/// legal input of CreateConcat function.
+/// \returns the dimension where the Concat will take place. If not
+/// concatenatable, return -1.
+static int getConcatDimensionForInputs(const std::vector<NodeValue>& inputs,
+                                       ConcatNode *origConcatN) {
+  // For the Concat to work, the inputs must only have one
+  // different dim, like <2,1,3>, <2,2,3> can be concatenated
+  // while <2,1,3>, <1,2,3> cannot. So for this function, it tries
```

Comment:
Since you have positive and negative cases, you may need to add more tests to check that the transform does not kick in in bad cases.

Reply:
I mean unit tests

Reply:
working on the unit tests